### ISSUE TITLE
Error parsing text

### ISSUE THREAD

**Issue by MrZillaGold** [PrismarineJS/mineflayer#1167]
## Versions
 - mineflayer: 2.24.0
 - server: Tuinity 1.15.2
 - node: 14.5.0

## Detailed description of a problem

When killing a player with a renamed item, an error occurs in the console
![image](https://user-images.githubusercontent.com/42776347/87894059-087b9780-ca85-11ea-85c9-4e5ee67f524a.png)

**Comment by Nurutomo** [comment]
Could you send the raw JSON or NBT tag to identify what is the cause of a
problem?

Pada tanggal Sen, 20 Jul 2020 09.34, Pronin Egor <notifications@github.com>
menulis:

**Comment by MrZillaGold** [comment]
![image](https://user-images.githubusercontent.com/42776347/88034833-94281d80-cb84-11ea-815f-de99cabab6a1.png)

It looks like the problem is in the lvl and count lines, they do not have a string type

**Comment by rom1504** [comment]
Yes can you provide this "json"

**Comment by MrZillaGold** [comment]
**1.15.2 Tuinity:**

```
Error parsing text '{id:"minecraft:diamond_pickaxe",tag:{display:{Name:'{"text":"Кирка хлебушка"}',Lore:['{"text":"MLG PICKAXE DORITOS 420NOSCOPEBLAZEIT"}','{"te
xt":"OH BABY A TRIPLE MOM GET THE CAMERA"}']},Enchantments:[{id:"minecraft:efficiency",lvl:20s},{id:"minecraft:fire_aspect",lvl:2s},{id:"minecraft:fortune",lvl:1
0s},{id:"minecraft:knockback",lvl:16959s},{id:"minecraft:looting",lvl:10s},{id:"minecraft:sharpness",lvl:16959s},{id:"minecraft:silk_touch",lvl:1s},{id:"minecraf
t:smite",lvl:16959s}],Unbreakable:1b,Damage:0,AttributeModifiers:[{Name:"generic.maxHealth",AttributeName:"generic.maxHealth",Operation:0,UUIDMost:0L,UUIDLeast:0
L,Amount:20.0d},{Name:"generic.movementSpeed",AttributeName:"generic.movementSpeed",Operation:0,UUIDMost:0L,UUIDLeast:0L,Amount:2.0d},{Name:"generic.knockbackRes
istance",AttributeName:"generic.knockbackResistance",Operation:0,UUIDMost:0L,UUIDLeast:0L,Amount:0.2d},{Name:"generic.attackDamage",AttributeName:"generic.attack
Damage",Operation:0,UUIDMost:0L,UUIDLeast:0L,Amount:0.0d}]},Count:1b}'
```

**1.16.1 Paper:**

`Error parsing text 'undefined'`

**Comment by ImHarvol** [comment]
I'm getting the same `Error parsing text 'undefined'` in mineflayer-web-inventory's tests: https://github.com/ImHarvol/mineflayer-web-inventory/runs/899820822. 
In my case it seems to happen when calling `bot.chat()`

**Comment by Karang** [comment]
This looks like the JSON-NBT format that is used in https://wiki.vg/Chat#Shared_between_all_components hover_event/show_item.
The format is also called SNBT https://minecraft.gamepedia.com/NBT_format
Do we have a library to parse it ?

**Comment by Karang** [comment]
Ok, I found node-mojangson. 

Looking at https://github.com/rom1504/node-mojangson/blob/master/grammar.jison it doesn't seems to handle russian characters, maybe that's the problem ?

**Comment by rom1504** [comment]
Pretty sure it does handle Russian character. The regex is just excluding
some meta characters

**Comment by rom1504** [comment]
If you feel like trying to add some more test cases to mojangson go ahead but tbh the definition of mojangson seems to be "json with everything lenient". It's hard to make it correct without some actual definition of it.

**Comment by rom1504** [comment]
With recent versions of mineflayer and prismarine-chat this kind of error is simply ignored and not displayed

**Comment by Karang** [comment]
Found these related libraries:
* https://github.com/Paril/lenientjsonjs
* https://github.com/rjrodger/jsonic
Maybe we can build something on top ?

**Comment by rom1504** [comment]
Maybe, I'd say first step would be to assemble a few examples that today do
not work with node-mojangson, see if there are any easy fix, otherwise try
to see if these libs can parse them.
Last time I looked at this, I found that the examples were really
unreasonable and I'm not even sure these lenient json libs could handle
them.
But it could be tried again

**Comment by Karang** [comment]
Fixed in node-mojangson 1.1.0
